sqlglot/generator.py

```diff
@@ -653,6 +653,9 @@ def hexstring_sql(self, expression: exp.HexString) -> str:
     def datatype_sql(self, expression: exp.DataType) -> str:
         type_value = expression.this
         type_sql = self.TYPE_MAPPING.get(type_value, type_value.value)
+        prefix_sql = expression.args.get("prefix")
+        prefix_sql = f"{prefix_sql}." if prefix_sql else ""
+        type_sql = f"{prefix_sql}{type_sql}"
```

Comment:
I think it might be better to put this logic only in teradata. Like:
```python
def datatype_sql(self, expression: exp.DataType) -> str:
    type_sql = super().datatype_sql(expression)
    prefix_sql = expression.args.get("prefix")
    return f"{prefix_sql}.{type_sql}" if prefix_sql else ""
```

Reply:
Also, if it's always `SYSUDTLIB` you could just use a boolean arg instead?

Reply:
Also, maybe we can just use a boolean arg for this, since SYSUDTLIB is the only supported prefix right now.